dev-docs/design/how-lit-works.md

```diff
@@ -0,0 +1,302 @@
+
+# Foundational Pieces
+
+lit-html relies on two key platform features for its performance and ergonomics: [JavaScript Tagged Template Literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals#tagged_templates) and the HTML [`<template>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/template) element.
+
+## JavaScript Tagged Template Literals
+
+lit-html's rendering process starts with defining a template with a [tagged template literal](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals#Tagged_templates).
+
+A _tagged_ template literal is essentially a function call that passes the strings literals and the values of expressions to the tag function.
+
+This code:
+
+```js
+let tag = (strings, ...values) => {
+  console.log('strings: ', strings);
+  console.log('values: ', values);
+};
+let text = 'abc';
+tag`<div>${text}</div>`;
+```
+
+prints:
+
+```js
+strings: ["<div>", "</div>", raw: Array(2)]
+values: ["abc"]
+```
+
+A tag function can return any object, it does not have to return a string.
+
+The first argument to the tag is a special array of strings that retains its identity across multiple evaluations of the tagged literal, and is always the exact same array of strings.
+
+We can see that the identity is preserved if we put the tagged literal in a function:
+
+```js
+// Store how many times we've seen each literal
+let counts = new Map();
+
+// A template tag that updates the count storage and longs the count
+let count = (strings) => {
+  // The strings object is our Map key
+  let c = (counts.get(strings) || 0) + 1;
+  counts.set(strings, c);
+  console.log(`literal '${strings.join('')}' seen ${c} time(s)`);
+};
+
+// Different invocations of this function return the same tagged template literal.
+let fn = (val) => count`static portion of template with a dynamic value:${val}`;
+
+// These two functions call the same tag function. Thus both `f` and `g` will increment
+// the count for the same template strings array even though the dynamic values are different.
+let f = () => fn('abc');
+let g = () => fn('def');
+
+f();
+g();
+```
+
+Prints:
+
+```bash
+literal 'static portion of template with a dynamic value:' seen 1 time(s)
+literal 'static portion of template with a dynamic value:' seen 2 time(s)
+```
+
+lit-html uses the template strings array argument as a cache key to only do template preparation work on the first render to the page.
+
+## The HTML `<template>` Element
+
+The [`<template>` element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/template) is a container for _inert_ HTML content. Its contents are parsed, but not rendered: scripts don't run, styles don't apply, Custom Elements don't upgrade, etc.
```

Comment:
Avoid using `inert` since it has s[pecific meaning ](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/inert)